### ISSUE TITLE
[macOS] voice over does not announce feedback when typing in a text field

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#77834]
```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';
void main() {
  runApp(
    MaterialApp(
      home: Scaffold(
          body: Scaffold(
              body: Center(
                  child: Column(
                    mainAxisAlignment: MainAxisAlignment.center,
                    children: [
                      TextField(controller: TextEditingController(text: 'thisistext'),)
                    ],
                  )))),
    ),
  );
}
```

1. in macOS, turn on voice over
2. move focus to textfield
3. the start typing

expect: the voice over should announce typed character
actual: nothing happened

**Comment by chunhtai** [comment]
When we post NSAccessibilityValueChangedNotification

This is the call stack
```
mshMIGPerform
  _XCopyParameterizedAttributeValue Line 443
    _AXXMIGCopyParameterizedAttributeValue
      CopyParameterizedAttributeValue
        NSAccessibilityEntryPointValueForAttributeWithParameter
          NSAccessibilityPerformEntryPointObject
            ___NSAccessibilityEntryPointValueForAttributeWithParameter_block_invoke
              [NSAccessibilityAttributeAccessorInfo getParameterizedAttributeValue:forObject:withParameter:]
                [NSAccessibilityAttributeAccessorInfo _getParameterizedAttributeForObject:withParameter:]
                  *[AXPlatformNodeCocoa accessibilityStyleRangeForIndex:] Line 83
                  *[AXPlatformNodeCocoa accessibilityAttributedStringForRange:] Line 87
  _XCopyAttributeValue Line 443
    _AXXMIGCopyAttributeValue
      CopyAttributeValue
        CopyAppKitUIElementAttributeValueNoCatch
          [NSObject(NSAccessibilityInternal) _accessibilityValueForAttribute:clientError:]
            _NSAccessibilityEntryPointValueForAttribute
              NSAccessibilityPerformEntryPointObject
                ___NSAccessibilityEntryPointValueForAttribute_block_invoke
                  [NSAccessibilityAttributeAccessorInfo getAttributeValue:forObject:]
                    [NSAccessibilityAttributeAccessorInfo _getAttributeValueForObject:]
                      *[AXPlatformNodeCocoa accessibilityVisibleCharacterRange] Line 76
                      *[AXPlatformNodeCocoa accessibilitySelectedTextRange] Line 76
                      *[AXPlatformNodeCocoa accessibilityPlaceholderValue] Line 76
```
functions with `*` are our code
mshMIGPerform is likely the main event handler it called _XCopyParameterizedAttributeValue and _XCopyAttributeValue at the same time probably due to one liner like
`[self doSomething:_XCopyParameterizedAttributeValue with:_XCopyAttributeValue];`
That is all the call stack that calls the accessibility function in our code.
